src/opcodes/opcodes.cairo

```diff
@@ -388,4 +389,14 @@ pub mod Opcode {
         let a = engine.astack.pop_byte_array();
         engine.dstack.push_byte_array(a);
     }
+
+    fn opcode_equal(ref engine: Engine) {
+        let a = engine.dstack.pop_int();
```

Comment:
For OP_EQUAL we should be comparing full byte arrays, not just integers. So we will want to use `pop_byte_array` here for instance.

Reply:
ooh, yeah, that makes sense.